syntax = "proto3";
package proto;
import "google/protobuf/any.proto";


enum ChatType {
    UNKNOWN = 0;
    USER = 1;
    GROUP = 2;
    BOT = 3;
}

message Content {
    string text = 1;
}

message Msg {
    Header header = 1;
    google.protobuf.Any data = 2;

    message Header {
        string type = 2;
    }
}

message PushMessage {
    string message_id = 1;
    Sender sender = 2;
    string chatId = 4;
    ChatType chatType = 5;
    Content content = 6;
    int64 contentType = 7;
    int64 time = 8;

    message Sender {
        string id = 1;
        ChatType chatType = 2;
        string nickname = 3;
    }
}

message SendMessage{
    string messageId = 2;
    string chatId = 3;
    ChatType chatType = 4;
    Content content = 5;
    int64 contentType = 6;
}

message GroupId {
    string groupId = 2;
}

message GroupInfo {
    Group group = 2;

    message Group {
        string name = 2;
    }
}